tests/epyccel/test_builtins.py

```diff
@@ -1,41 +1,152 @@
+    def f(x, y):
+        return min(x, y)
+
+    a = randint(100)
+    b = randint(100)
+    epyc_f = epyccel(f, language=language)
+    assert epyc_f(a,b) == f(a,b)
+    assert epyc_f(float(a),float(b)) == f(float(a),float(b))
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = pytest.mark.fortran),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="min not implemented in C"),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+def test_min_3_args(language):
+    @types('int','int','int')
+    @types('float','float','float')
+    def f(x, y, z):
+        return min(x, y, z)
+
+    a = randint(100)
+    b = randint(100)
+    c = randint(100)
+    epyc_f = epyccel(f, language=language)
+    assert epyc_f(a,b,c) == f(a,b,c)
+    assert epyc_f(float(a),float(b),float(c)) == f(float(a),float(b),float(c))
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = pytest.mark.fortran),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="max not implemented in C"),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+def test_max_2_args(language):
+    @types('int','int')
+    @types('float','float')
+    def f(x, y):
+        return max(x, y)
+
+    a = randint(100)
+    b = randint(100)
+    epyc_f = epyccel(f, language=language)
+    assert epyc_f(a,b) == f(a,b)
+    assert epyc_f(float(a),float(b)) == f(float(a),float(b))
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = pytest.mark.fortran),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="max not implemented in C"),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+def test_max_3_args(language):
+    @types('int','int','int')
+    @types('float','float','float')
+    def f(x, y, z):
+        return min(x, y, z)
+
+    a = randint(100)
+    b = randint(100)
+    c = randint(100)
+    epyc_f = epyccel(f, language=language)
+    assert epyc_f(a,b,c) == f(a,b,c)
+    assert epyc_f(float(a),float(b),float(c)) == f(float(a),float(b),float(c))
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = pytest.mark.fortran),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="sum not implemented in C"),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+def test_sum_matching_types(language):
+    @template('T',['int','float','complex'])
+    @types('T','T')
+    def f(x, y):
+        return sum([x, y])
+
+    a = randint(100)
+    b = randint(100)
+    epyc_f = epyccel(f, language=language)
+    assert epyc_f(a,b) == f(a,b)
+    assert epyc_f(float(a),float(b)) == f(float(a),float(b))
+    assert epyc_f(complex(a),complex(b)) == f(complex(a),complex(b))
```

Comment:
same here, generate random values is better than casting.

Reply:
See 17d3c40